pyHS100/pyHS100.py

```diff
@@ -534,3 +279,477 @@ def mac(self, mac):
+            return False
+
+        response = self._query_helper(self.emeter_type, "get_monthstat",
+                                      {'year': year})
+
+        if self.emeter_units:
+            key = 'energy_wh'
+        else:
+            key = 'energy'
+
+        return {entry['month']: entry[key]
+                for entry in response['month_list']}
+
+    def erase_emeter_stats(self):
+        """
+        Erase energy meter statistics
+
+        :return: True if statistics were deleted
+                 False if device has no energy meter.
+        :rtype: bool
+        :raises SmartPlugException: on error
+        """
+        if not self.has_emeter:
+            return False
+
+        self._query_helper(self.emeter_type, "erase_emeter_stat", None)
+
+        # As query_helper raises exception in case of failure, we have
+        # succeeded when we are this far.
+        return True
+
+    def current_consumption(self):
+        """
+        Get the current power consumption in Watt.
+
+        :return: the current power consumption in Watt.
+                 False if device has no energy meter.
+        :raises SmartPlugException: on error
+        """
+        if not self.has_emeter:
+            return False
+
+        response = self.get_emeter_realtime()
+        if self.emeter_units:
+            return response['power_mw']
+        else:
+            return response['power']
+
+
+class SmartBulb(SmartDevice):
+    """Representation of a TP-Link Smart Bulb.
+
+    Usage example when used as library:
+    p = SmartBulb("192.168.1.105")
+    # print the devices alias
+    print(p.alias)
+    # change state of bulb
+    p.state = "ON"
+    p.state = "OFF"
+    # query and print current state of plug
+    print(p.state)
+    # check whether the bulb supports color changes
+    if p.is_color:
+    # set the color to an HSV tuple
+    p.hsv = (100, 0, 255)
+    # get the current HSV value
+    print(p.hsv)
+    # check whether the bulb supports setting color temperature
+    if p.is_variable_color_temp:
+    # set the color temperature in Kelvin
+    p.color_temp = 3000
+    # get the current color temperature
+    print(p.color_temp)
+    # check whether the bulb is dimmable
+    if p.is_dimmable:
+    # set the bulb to 50% brightness
+    p.brightness = 50
+    # check the current brightness
+    print(p.brightness)
+
+    Errors reported by the device are raised as SmartPlugExceptions,
+    and should be handled by the user of the library.
+
+    """
+    # bulb states
+    BULB_STATE_ON = 'ON'
+    BULB_STATE_OFF = 'OFF'
+
+    def __init__(self, ip_address, protocol=None):
+        SmartDevice.__init__(self, ip_address, protocol)
+        self.is_color = self.sys_info['is_color']
+        self.is_dimmable = self.sys_info['is_dimmable']
+        self.is_variable_color_temp = self.sys_info['is_variable_color_temp']
+        self.emeter_type = "smartlife.iot.common.emeter"
+        self.emeter_units = True
+
```

Comment:
That's true, #26 changed the behaviour to disable caching of responses from the bulb so that every access to sys_info will trigger a request. I'm having second thoughts if it'd make more sense to revert that and make clients to adapt to #20.